Packs/EmailCommunication/Scripts/PreprocessEmail/PreprocessEmail.py

```diff
@@ -222,6 +252,9 @@ def main():
 
     except (IndexError, ValueError, DemistoException) as e:
         # True - For creating new incident
+        demisto.executeCommand('setIncident', {'id': incident.get('id'),
+                                               'customFields': {'emailcodegenerator': get_unique_code()}})
+
         demisto.results(True)
         return_error(f"The PreprocessEmail script has encountered an error:\n {e} \nA new incidents was created.")
```

Comment:
this whole handling is strange.
we are catching a family if problems where the cases we do not find the current incident(if it is new) should be much more narrow. then we log it as an error that was encountered where it include the flow that is expected and not an error at all.